src/wp-includes/rest-api.php

```diff
@@ -1254,6 +1255,41 @@ function rest_add_application_passwords_to_index( $response ) {
 	return $response;
 }
 
+/**
+ * Adds the default template types and template part areas to the REST API index.
+ *
+ * @since 6.8.0
+ *
+ * @param WP_REST_Response $response REST API response.
+ * @return WP_REST_Response Modified REST API response.
+ */
+function rest_add_templates_default_data_to_index( $response ) {
+	$can_edit_posts = current_user_can( 'edit_posts' );
```

Comment:
I think it's needed in the site editor so theme editors should be able to see it.

```suggestion
	$can_edit_posts = current_user_can( 'edit_posts' ) || current_user_can( 'edit_theme_options' );
```

Reply:
This is a common pattern in REST endpoints. I'm not sure if the extra `edit_theme_options` check is needed.

The site editor would be pretty useless without the `edit_posts` cap.

https://github.com/WordPress/wordpress-develop/blob/2a0517b05f0ba2a30de1e8153836cdbf942b7c29/src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php#L194-L201